datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -37,6 +37,22 @@ def backend_normalize_metric_name(metric_name):
     return METRIC_DOTUNDERSCORE_CLEANUP.sub(".", metric_name).strip("_")
 
 
+def check_tag_names(metric, tags):
+    forbidden_tags = [
+        'cluster_name',
+        'clustername',
+        'cluster',
+        'host_name',
+        'hostname',
+        'host',
+    ]
+
+    for tag in tags:
+        tag_name = tag.split(':')[0]
+        if  tag_name in forbidden_tags:
+            raise Exception("Metric {} was submitted with a forbidden tag: {}".format(metric, tag_name))
```

Comment:
Raising an exception for now to see how many integrations are impacted.
Some of these integration are already "fixed", with a configuration option to disable these generic tags (ex: spark).